chirpstack/src/codec/mod.rs

```diff
@@ -35,17 +39,28 @@ where
     *const str: deserialize::FromSql<Text, DB>,
 {
     fn from_sql(value: <DB as Backend>::RawValue<'_>) -> deserialize::Result<Self> {
-        let string = String::from_sql(value)?;
-        Ok(Codec::from_str(&string)?)
+        let string = <*const str>::from_sql(value)?;
+        Ok(Self::from_str(unsafe { &*string })?)
```

Comment:
Diesel doesn't have a `FromSql` impl for &str [due to internal constraints](https://docs.rs/diesel/2.1.6/diesel/deserialize/trait.FromSql.html#impl-FromSql%3CText,+Sqlite%3E-for-*const+str) so we have 2 choices:
- use `FromSql` impl on `*const str` which needs unsafe
- use `FromSql` impl on String which [needs an extra allocation](https://github.com/diesel-rs/diesel/blob/master/diesel/src/type_impls/primitives.rs#L126) and does the same as here.

I preferred first option since it's recommended by diesel and is more optimized.

Reply:
As agreed, the current implementation is ok to merge, we can see later if it needs changes.